tests/integration/targets/aci_switch_access_config/tasks/main.yml

```diff
@@ -0,0 +1,222 @@
+# Test code for the ACI modules
+# Copyright: (c) 2025, Samita Bhattacharjee (@samiib) <samitab@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  ansible.builtin.fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+# SET VARS
+- name: Set vars
+  ansible.builtin.set_fact:
+    aci_info: &aci_info
+      host: '{{ aci_hostname }}'
+      username: '{{ aci_username }}'
+      password: '{{ aci_password }}'
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("debug") }}'
+
+- name: Query system information
+  cisco.aci.aci_system:
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  ansible.builtin.include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for ACI v6+ and non-cloud sites
+  when:
+  - version.current.0.topSystem.attributes.version is version('6', '>=')
+  - query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
```

Comment:
This is not required to change and just a suggestion for your consideration. We have not done thus far, but I have been looking into block documentation for NDO testing: https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_blocks.html. We could consider putting the # CLEAN ENVIRONMENT path at the end in a always section. This way even when the execution fails halfway through the test for whatever reason we can attempt to clean up the config. The drawback would be that troubleshooting might be more complex if you want to look at the state of the configuration after fail scenario.